src/VectorizeLoops.cpp

```diff
@@ -839,7 +858,11 @@ class VectorSubs : public IRMutator {
                 // *every* likely value is true. We can do that by
                 // generating a scalar condition that checks if
                 // the least-true lane is true.
-                Expr all_true = bounds_of_lanes(likely->args[0]).min;
+                Expr all_true = likely->args[0];
+                while (!all_true.type().is_scalar()) {
```

Comment:
Wait, what? bounds_of_lanes only unpeels one layer of vector? We should just fix it instead of having a while loop around it.